### ISSUE TITLE
Jetpack.com: homepage has missing headings

### ISSUE THREAD

**Issue by jkguidaven** [Automattic/jetpack#30187]
## What's the issue
The Jetpack.com homepage skips heading levels: the product card sections include `h3` tags but don't have an `h2` tag. This is an accessibility (and maybe SEO) issue.

## How to reproduce the issue
- Visit https://jetpack.com/
- Open a screen reader of your choice, or open the dev console
- Notice the product cards sections have `h3` headings but not `h2`

## Expected behavior
Product card sections should have a level 2 heading (e.g., `Popular products`), which should at least be made visible to screen readers.

## Captures
_Product card section_
<img width="400" src="https://user-images.githubusercontent.com/1620183/229602920-b5f88307-75c5-444d-9a99-7e871bd4cef8.png" alt=""/>

_Document outline shown by VoiceOver_
<img width="400" src="https://user-images.githubusercontent.com/1620183/229602648-50ff7ab2-bc7d-42a4-9f9b-8e8974b6ec28.png" alt=""/>

**Comment by jkguidaven** [comment]
diff: D108742-code

**Comment by jkguidaven** [comment]
Diff has been deployed
